app/upload/upload-file-select.directive.ts

```diff
@@ -0,0 +1,22 @@
+import {Directive, ElementRef, EventEmitter} from 'angular2/core';
+import {UploadService} from './upload.service';
+
+@Directive({
+  selector: '[upload-file]',
+  inputs: ['uploadService'],
```

Comment:
That may be an old way to specify an @Input, that was in the docs when I looked before, but it is gone now - I'll try updating it to the annotation style.